src/backends/gpu/matrix_state_interleaved.hpp

```diff
@@ -0,0 +1,255 @@
+#pragma once
```

Comment:
Yes.

I have cleaned it up, but I do like the style

```
//
// Stage X
///
```
to mark out regions in longer functions.